### PR TITLE
ci(github-actions): add Prettier auto-correct for lint issues on PRs

### LINKED ISSUE #711: [FEATURE] add prettier to github actions
We need to add prettier GitHub actions, so when a PR is raised it auto corrects any lint issues

> Note: Please read the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by joshua-jinu**
# Fixes Issue

**My PR closes #711**

# 👨‍💻 Changes proposed(What did you do ?)
- Addition of formatting using Prettier and ESLint on Pull Requests using Github Actions.

# ✔️ Check List (Check all the applicable boxes)

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

# 📷 Screenshots

### CODE REVIEW

**Comment by Dun-sin** [comment]
@joshua-jinu your lint fails

**Comment by Dun-sin** [comment]
did you use chatgpt? looks like you didn't study the project structure to do this

File: .github/workflows/prettier-eslint.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        with:
+          node-version: "16"
+
+      - name: Install Dependencies
+        run: npm install
+
+      - name: Run ESLint and Prettier
+        run: |
+          npm run lint -- --fix
+          npm run format -- --write
```
Comment (Dun-sin) [inline .github/workflows/prettier-eslint.yml:28]: no need for the extra options:
https://github.com/Dun-sin/Whisper/blob/c2dc613d32e1cfb41e11098db3bfe3cc19c2e723/client/package.json#L9-L10

File: .github/workflows/prettier-eslint.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+
+          git add .
+          git commit -m "chore: auto-fix lint and prettier issues" || echo "No changes to commit"
+          git push origin feature/prettier-github-action
+
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Dun-sin) [inline .github/workflows/prettier-eslint.yml:40]: don't think any of this is important, try taking it out

**Review by Dun-sin — CHANGES_REQUESTED** [review]

**Comment by joshua-jinu** [comment]
Hey @Dun-sin , I've made the changes, could you please review them

File: .github/workflows/prettier-eslint.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: "20.x"
+
+      - name: Install Dependencies
+        run: |
+          cd ../../server && npm i
+          cd ../client && npm i
```
Comment (Dun-sin) [inline .github/workflows/prettier-eslint.yml:25]:
```suggestion
          cd ./server && npm i
          cd ../client && npm i
```

File: .github/workflows/prettier-eslint.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: "20.x"
+
+      - name: Install Dependencies
+        run: |
+          cd ./server && npm i
+          cd ../client && npm i
```
Comment (Dun-sin) [inline .github/workflows/prettier-eslint.yml:25]:
```suggestion
          cd ../client && npm i --force
```

File: .github/workflows/prettier-eslint.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          node-version: "20.x"
+
+      - name: Install Dependencies
+        run: |
+          cd ./server && npm i
+          cd ../client && npm i --force
+
+      - name: Run ESLint and Prettier
+        run: |
```
Comment (Dun-sin) [inline .github/workflows/prettier-eslint.yml:28]:
```suggestion
      - name: Run ESLint and Prettier
        working-directory: ./client
        run: |
```

File: .github/workflows/prettier-eslint.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Install Dependencies
+        run: |
+          cd ./server && npm i
+          cd ../client && npm i --force
+
+      - name: Run ESLint and Prettier
+        working-directory: ./client
+        run: |
+          npm run lint
+          npm run format
```
Comment (Dun-sin) [inline .github/workflows/prettier-eslint.yml:31]:
```suggestion
name: Prettier Lint and Format
on:
  pull_request:
    branches:
      - main
permissions:
  contents: write
  pull-requests: write
jobs:
  prettier-lint-and-format:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout Code
        uses: actions/checkout@v3
        with:
          ref: ${{ github.head_ref }}
          token: ${{ secrets.GITHUB_TOKEN }}
      - name: Set up Node.js
        uses: actions/setup-node@v3
        with:
          node-version: "20.x"
      - name: Install Dependencies
        run: |
          cd ./server && npm i
          cd ../client && npm i --force
      - name: Run ESLint and Prettier
        working-directory: ./client
        run: |
          npm run lint
          npm run format
      - name: Check for changes
        id: git-check
        run: |
          git diff --exit-code || echo "has_changes=true" >> $GITHUB_OUTPUT
      - name: Commit changes
        if: steps.git-check.outputs.has_changes == 'true'
        run: |
          git config --local user.email "github-actions[bot]@users.noreply.github.com"
          git config --local user.name "github-actions[bot]"
          git commit -am "style: format code with prettier and fix lint issues"
          git push
```

File: .github/workflows/prettier-eslint.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      - main
+permissions:
+  contents: write
+  pull-requests: write
+jobs:
+  prettier-lint-and-format:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
```
Comment (Dun-sin) [inline .github/workflows/prettier-eslint.yml:14]:
```suggestion
        uses: actions/checkout@v2
```

File: .github/workflows/prettier-eslint.yml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        id: git-check
+        run: |
+          git diff --exit-code || echo "has_changes=true" >> $GITHUB_OUTPUT
+      - name: Commit changes
+        if: steps.git-check.outputs.has_changes == 'true'
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git commit -am "style: format code with prettier and fix lint issues"
+          git push
```
Comment (Dun-sin) [inline .github/workflows/prettier-eslint.yml:41]:
```suggestion
        uses: actions/checkout@v3
        with:
          fetch-depth: 0
          ref: ${{ github.event.pull_request.head.ref }}
          repository: ${{ github.event.pull_request.head.repo.full_name }}
          token: ${{ secrets.GITHUB_TOKEN }}
      - name: Set up Node.js
        uses: actions/setup-node@v3
        with:
          node-version: "20.x"
      - name: Install Dependencies
        run: |
          cd ./server && npm i
          cd ../client && npm i --force
      - name: Run ESLint and Prettier
        working-directory: ./client
        run: |
          npm run lint
          npm run format
      - name: Check for changes
        id: git-check
        run: |
          git diff --quiet || echo "has_changes=true" >> $GITHUB_OUTPUT
      - name: Commit changes
        if: steps.git-check.outputs.has_changes == 'true'
        run: |
          git config --local user.email "github-actions[bot]@users.noreply.github.com"
          git config --local user.name "github-actions[bot]"
          git commit -am "style: format code with prettier and fix lint issues"
          git add -A
          git push
```

File: .github/workflows/prettier-eslint.yml [hunk 9]
```diff
@@ -0,0 +1,44 @@
+name: Prettier Lint and Format
+on:
+  pull_request:
+    branches:
+      - main
+permissions:
+  contents: write
+  pull-requests: write
```
Comment (Dun-sin) [inline .github/workflows/prettier-eslint.yml:8]:
```suggestion
  push:
    branches:
      - main
      
permissions:
  contents: write
```

**Review by Dun-sin — APPROVED** [review]
You did great, thanks for contributing, I hope you will stick around and continue to contribute to this project.

Consider giving this project a star, sharing the project with your friends, and joining the community discord server if you haven't for more resources and opportunities to connect with others. [👉🏽here👈🏽](https://discord.com/invite/ufcysW9q23)

**Comment by joshua-jinu** [comment]
Thank you @Dun-sin, looking forward to work on more such PRs

**Pull request merged**